tokio-trace/test_static_max_level_features/main.rs

```diff
@@ -0,0 +1,75 @@
+#[macro_use]
+extern crate log;
+
+use std::sync::{Arc, Mutex};
+use log::{Level, LevelFilter, Log, Record, Metadata};
+
+#[cfg(feature = "std")]
+use log::set_boxed_logger;
+#[cfg(not(feature = "std"))]
+fn set_boxed_logger(logger: Box<Log>) -> Result<(), log::SetLoggerError> {
+    unsafe {
+        log::set_logger(&*Box::into_raw(logger))
+    }
+}
+
+struct State {
+    last_log: Mutex<Option<Level>>,
+}
+
+struct Logger(Arc<State>);
```

Comment:
Since this will become a `tokio-trace` `Subscriber` instead of a `Logger`, we'll probably want to rename it to something like 
```suggestion
struct TestSubscriber(Arc<State>);
```